### ISSUE TITLE
How to fine-tune BERT on a large training dataset?

### ISSUE THREAD

**Issue by wcgan** [huggingface/transformers#1867]
## ❓ Questions & Help

Hi,

I want to fine-tune BERT on a large training dataset. With around 1.5million training examples, this is currently consuming around 60GB of RAM. Is there any way to reduce the RAM usage or load the training examples in parts? Thanks!.

**Comment by iedmrc** [comment]
I think you should check out pytorch dataloaders https://pytorch.org/docs/stable/data.html
Also, gradient accumulation is helpful.
